File: Python/person.py
from abc import ABC, abstractmethod

# Implemented Abstraction

class Person(ABC):

    def __init__(self, id, name):
        self.__id = id
        self.__name = name

    # Implemented Encapsulation
    def getId(self):
        return self.__id

    def getName(self):
        return self.__name

    def setId(self, id):
        self.__id = id

    def setName(self, name):
        self.__name = name

    @abstractmethod
    def showRole(self):
        pass

File: C#/Person.cs
using System;

namespace StudentManagementSystem
{
    // Implemented Abstraction
    public abstract class Person
    {
        private int id;
        private string name;

        protected Person(int id, string name)
        {
            this.id = id;
            this.name = name;
        }

        // Implemented Encapsulation
        public int Id
        {
            get { return id; }
            set { id = value; }
        }

        public string Name
        {
            get { return name; }
            set { name = value; }
        }

        public abstract void ShowRole();
    }
}
